src/tests/class_tests/openms/source/TargetedSpectraExtractor_test.cpp

```diff
@@ -0,0 +1,837 @@
+    << "\t 1/fwhm: " << f.getMetaValue("inverse_avgFWHM")
+    << "\t SNR: " << f.getMetaValue("avgSNR")
+    << "\t fwhm: " << f.getMetaValue("avgFWHM"))
+  }
+}
+END_SECTION
+
+START_SECTION(selectSpectra())
+{
+  ptr->setUseGauss(true);
+  ptr->setGaussWidth(0.25);
+  ptr->setRTWindow(30.0);
+  ptr->setMZTolerance(0.1);
+  ptr->setPeakHeightMin(15000.0);
+  ptr->setPeakHeightMax(110000.0);
+  ptr->setFWHMThreshold(0.23);
+  ptr->setTICWeight(1.0);
+  ptr->setFWHMWeight(1.0);
+  ptr->setSNRWeight(1.0);
+  ptr->setMinScore(15.0);
+
+  const std::vector<MSSpectrum> spectra = experiment.getSpectra();
+  std::vector<MSSpectrum> annotated;
+  FeatureMap features;
+  ptr->annotateSpectra(spectra, targeted_exp, annotated, features);
+  std::vector<MSSpectrum> picked(annotated.size());
+  for (UInt i=0; i<annotated.size(); ++i)
+  {
+    ptr->pickSpectrum(annotated[i], picked[i]);
+  }
+  for (Int i=annotated.size()-1; i>=0; --i)
+  {
+    if (!picked[i].size())
+    {
+      annotated.erase(annotated.begin() + i);
+      picked.erase(picked.begin() + i);
+      features.erase(features.begin() + i);
+    }
+  }
+  std::vector<MSSpectrum> scored;
+  ptr->scoreSpectra(annotated, picked, features, scored);
+
+  STATUS("Scored spectra and their score:")
+  for (auto s : scored)
+  {
+    STATUS(s.getName() << "\t" << s.getFloatDataArrays()[1][0])
+  }
+
+  std::vector<MSSpectrum> selected_spectra;
+  FeatureMap selected_features;
+
+  ptr->selectSpectra(scored, features, selected_spectra, selected_features);
+  TEST_EQUAL(selected_spectra.size(), 2)
+  TEST_EQUAL(selected_spectra.size(), selected_features.size())
+  for (UInt i=0; i<selected_spectra.size(); ++i)
+  {
+    TEST_NOT_EQUAL(selected_spectra[i].getName(), "")
+    TEST_EQUAL(selected_spectra[i].getName(), selected_features[i].getMetaValue("transition_name"))
+    TEST_EQUAL(selected_spectra[i].getFloatDataArrays()[1][0], selected_features[i].getIntensity())
+    TEST_EQUAL(selected_spectra[i].getFloatDataArrays()[1][0] >= ptr->getMinScore(), true)
+  }
+
+  ptr->selectSpectra(scored, selected_spectra);
+  TEST_NOT_EQUAL(selected_spectra.size(), 0)
+  for (UInt i=0; i<selected_spectra.size(); ++i)
+  {
+    TEST_NOT_EQUAL(selected_spectra[i].getName(), "")
+    TEST_EQUAL(selected_spectra[i].getFloatDataArrays()[1][0] >= ptr->getMinScore(), true)
+  }
+}
+END_SECTION
+
+START_SECTION(extractSpectra())
+{
+  ptr->setUseGauss(true);
+  ptr->setGaussWidth(0.25);
+  ptr->setRTWindow(30.0);
+  ptr->setMZTolerance(0.1);
+  ptr->setPeakHeightMin(15000.0);
+  ptr->setPeakHeightMax(110000.0);
+  ptr->setFWHMThreshold(0.23);
+  ptr->setTICWeight(1.0);
+  ptr->setFWHMWeight(1.0);
+  ptr->setSNRWeight(1.0);
+  ptr->setMinScore(15.0);
+
+  vector<MSSpectrum> extracted_spectra;
+  FeatureMap extracted_features;
+  ptr->extractSpectra(experiment, targeted_exp, extracted_spectra, extracted_features);
+
+  TEST_EQUAL(extracted_spectra.size(), extracted_features.size())
```

Comment:
again, test for equality with a number, to make sure that the algorithm produces the same output also in the future. again, testing the first spectrum may not be a bad idea either